CRM/Batch/Form/Entry.php

```diff
@@ -1237,19 +1231,17 @@ public static function legacyProcessMembership($contactID, $membershipTypeID, $i
     }
     $params['modified_id'] = $modifiedID ?? $contactID;
 
-    $memParams['contribution'] = $contribution;
     $memParams['custom'] = $customFieldsFormatted;
     // Load all line items & process all in membership. Don't do in contribution.
     // Relevant tests in api_v3_ContributionPageTest.
-    $memParams['line_item'] = $lineItems;
     // @todo stop passing $ids (membership and userId may be set by this point)
     $membership = CRM_Member_BAO_Membership::create($memParams, $ids);
 
     // not sure why this statement is here, seems quite odd :( - Lobo: 12/26/2010
     // related to: http://forum.civicrm.org/index.php/topic,11416.msg49072.html#msg49072
     $membership->find(TRUE);
 
-    return [$membership, $renewalMode, $dates];
+    return $membership;
```

Comment:
Grepped `universe` for references to `legacyProcessMembership` - nothing outside of `civicrm-core` using it.

@eileenmcnaughton Just to double-check, the `legacyProcessMembership` in `CRM/Contribute/Form/Contribution/Confirm.php` and `CRM/Batch/Form/Entry.php` are essentially independent, right? So it looks like this change to the `Batch/Form/Entry.php` has no interaction with the `Confirm.php` one.

If so, I'd say go ahead and merge.

Reply:
I believe we have unit testes that cover the Confirm.php stuff as well